### ISSUE TITLE
Adopt Nullable Reference Types

### ISSUE THREAD

**Issue by vany0114** [Polly-Contrib/Simmy#54]
Throwing this here to take into consideration, according to Microsoft, we should consider adopting NRT by next year:
https://devblogs.microsoft.com/dotnet/embracing-nullable-reference-types/

There are some caveats tho regarding C# 8 and multitargeting that I'm not sure if we have to care about (I have to dig into it) but as I said, just to consider it, let me know your thoughts @reisenberger please.

**Comment by reisenberger** [comment]
Yes, it would be good for any library like Simmy to adopt this by Nov 2020.

Given Simmy extends Polly, it probably makes sense to implement nullable reference types for Simmy after it has been implemented for Polly, as some settings for Simmy will 'flow' from attribution/NRT-decisions made on the external surface of Polly.

>caveats regarding C# 8 and multitargeting that I'm not sure if we have to care about

Where we need to apply attributes that are not part of a given target by default, we can probably adopt the strategy that those targets manually include the attribute code (probably the same thing you have read about).
